Sources/BraintreeVisaCheckout/BTVisaCheckoutClient.swift

```diff
@@ -116,14 +107,20 @@ import BraintreeCore
         encryptedPaymentData: String?,
         completion: @escaping (BTVisaCheckoutNonce?, Error?) -> Void
     ) {
-        if statusCode == .statusUserCancelled {
-            notifyFailure(with: BTVisaCheckoutError.canceled, completion: completion)
-            return
-        }
-
-        guard statusCode == .statusSuccess else {
-            notifyFailure(with: BTVisaCheckoutError.checkoutUnsuccessful, completion: completion)
-            return
+        if statusCode != .statusSuccess {
+            switch statusCode {
+            case .statusDuplicateCheckoutAttempt, .statusNotConfigured, .statusInternalError:
+                notifyFailure(with: BTVisaCheckoutError.checkoutUnsuccessful, completion: completion)
+                return
+            case .statusUserCancelled:
+                notifyFailure(with: BTVisaCheckoutError.canceled, completion: completion)
+                return
+            case .statusNetworkError:
+                notifyFailure(with: BTVisaCheckoutError.failedToCreateNonce, completion: completion)
+                return
+            default:
+                break
```

Comment:
should we consider a new error for this:
```suggestion
                notifyFailure(with: BTVisaCheckoutError.unknownStatus, completion: completion)
```